Jenkins jobs for antrea (#294)
vmware-tanzu/antrea: ci/jenkins/README.md

## Antrea CI: Jenkins

### Reasons for Jenkins
We have tests as Github Action but Jenkins allows tests running on a cluster of multiple nodes and offers better environment setup options.

> 

Reviewer: I think it would be great to include a list of the 3 Jenkins jobs we have here, with a brief description of each one. I would also include the list of the community tests we run in each job (or if the list is large / dynamic, which filters we use to select the community tests we want to run). I know the information is available in the yaml files, but it takes some time to find and I think it belongs in this README as well.

Author: Yes, I agree. Added "List of Jenkins jobs".

### Requirements

> Yaml files under ```ci/jenkins/jobs``` can be generated via jenkins-job-builder. If you want to try out the tests on your local jenkins setup, please notice the following requirements:

Reviewer: make ci/jenkins/jobs a markdown link maybe?

Author: Updated

* Jenkins setup
  * Plugins: ghprb, throttle-concurrents
* Install [jenkins-job-builder](https://docs.openstack.org/infra/jenkins-job-builder/index.html)

> * Define your ```ANTREA_GIT_CREDENTIAL``` which is the credential for your private repo

Reviewer: you don't need triple backticks for inline code highlighting, just a single one.

Author: Updated.

## Antrea CI: Jenkins

### Reasons for Jenkins
We have tests as Github Action but Jenkins allows tests running on a cluster of multiple nodes and offers better environment setup options.

### List of Jenkins jobs
* [e2e](https://github.com/vmware-tanzu/antrea/tree/master/test/e2e): end-to-end tests for Antrea.

> * conformance: community tests using sonobuoy, focusing on "Conformance" and skip "Slow", "Serial", "Disruptive", "Flaky", "Feature", "sig-cli", "sig-storage", "sig-auth", "sig-api-machinery", "sig-apps", or "sig-node".

Reviewer: s/skip/skipping

Author: changed "skip" to "skipping"

> * network policy: community tests using sonobuoy, focusing on "Feature:NetworkPolicy" and skip "allow ingress access from updated pod" or "named port".

Reviewer: s/skip/skipping
Please add a pointer to the named port Antrea issue (https://github.com/vmware-tanzu/antrea/issues/122) to explain why we are skipping "named port".
Please add a pointer to https://github.com/kubernetes/kubernetes/issues/85908 to explain why we are skipping "allow ingress access from updated pod"

Author: changed "skip" to "skipping". Two links added. Thanks.

## Antrea CI: Jenkins

### Reasons for Jenkins
We have tests as Github Action but Jenkins allows tests running on a cluster of multiple nodes and offers better environment setup options.

### List of Jenkins jobs
* [e2e](https://github.com/vmware-tanzu/antrea/tree/master/test/e2e): end-to-end tests for Antrea.
* conformance: community tests using sonobuoy, focusing on "Conformance" and skipping "Slow", "Serial", "Disruptive", "Flaky", "Feature", "sig-cli", "sig-storage", "sig-auth", "sig-api-machinery", "sig-apps", or "sig-node".

> * network policy: community tests using sonobuoy, focusing on "Feature:NetworkPolicy" and skipping "allow ingress access from updated pod" or "named port".  

Reviewer: ```suggestion
* network policy: community tests using sonobuoy, focusing on "Feature:NetworkPolicy", and skipping "allow ingress access from updated pod" and "named port". 
```

> Test "named port" will be supported soon, issue [#122](https://github.com/vmware-tanzu/antrea/issues/122).  

Reviewer: ```suggestion
Test "named port" will be supported soon, see issue [#122](https://github.com/vmware-tanzu/antrea/issues/122).
```

> Test "allow ingress access from updated pod" is a bug to be fixed, issue [#85908](https://github.com/kubernetes/kubernetes/issues/85908).

Reviewer: ```suggestion
Test "allow ingress access from updated pod" fails because of a bug in the test definition, see issue [#85908]
```

Author: I am not sure why you delete the link to kubernetes issue but keep the "[]" around "#85908" ?

Reviewer: I think that's just a typo on my part or something weird with the Github UI.

Reviewer: @lzhecheng I think you should put the link back, it was a typo on my part. I did not copy the whole line when I created the suggestion.

Author: OK, I have put back the link.
